src/service.ts

```diff
@@ -220,8 +223,12 @@ export class Service {
     delete reqOpts.interceptors_;
 
     const pkg = this.packageJson;
+    let userAgent = util.getUserAgentFromPackageJson(pkg);
+    if (this.providedUserAgent) {
+      userAgent = `${this.providedUserAgent} ${userAgent}`;
+    }
     reqOpts.headers = extend({}, reqOpts.headers, {
-      'User-Agent': util.getUserAgentFromPackageJson(pkg),
+      'User-Agent': userAgent,
```

Comment:
@dazuma @bcoe just to make sure - setting user agent like this won't have side effects for our own reporting will it?  All of our metrics and such go through `x-goog-api-client` headers, correct?

Reply:
The backend does still pick out data from both headers and tries to sort through it (because lots of legacy libraries), but it should be fine.